invrec_deutanal.py

```diff
@@ -0,0 +1,290 @@
+matplotlib.rcParams["figure.figsize"] = [8.0,5.0]
+#baseline fitting
+with figlist_var() as fl:
+    def calc_baseline(this_d,
+            ph1lim,
+            npts=5,
+            guess=None,
+            fl=None):
+        if show_plots: fl.next('try baseline correction')
+        if show_plots: fl.plot(this_d,
+                label='before')
+        this_d_tdom = this_d.C.ift('t2')
+        ndshape(this_d_tdom.alloc().copy_props(this_d_tdom)
+    def vec_to_params(ini_vec):
+            phcorr0,phcorr1 = ini_vec[:2]
+            baseline_vec = ini_vec[2:].view(complex128)
+            return phcorr0,phcorr1,baseline_vec
+        def apply_corr(ini_vec):
+            phcorr0,phcorr1,baseline_vec = vec_to_params(ini_vec)
+            d_test = this_d.C
+            d_test *= exp(-1j*phcorr1*d.fromaxis('t2')-1j*phcorr0)
+            d_test.ift('t2')
+            retval = d_test['t2',0:len(baseline_vec)] + baseline_vec
+            d_test['t2',0:len(baseline_vec)] = retval
+            return d_test.ft('t2')
+        def generate_baseline(baseline_vec):
+            baseline_data = blank_tdom.C
+            baseline_data['t2',0:len(baseline_vec)] += baseline_vec
+            return baseline_data
+        def costfun(ini_vec):
+            d_test = apply_corr(ini_vec)
+            return abs(d_test.real).sum('t2').data.item()
+        max_val = abs(this_d_tdom.data).max()
+        logger.info(strm(max_val))
+        mybounds = r_[-max_val,max_val][newaxis,:]*ones(npts*2)[:,newaxis]
+        print(shape(mybounds))
+        print(mybounds)
+        mybounds = r_[
+                r_[-pi,pi,-ph1lim,ph1lim].reshape(-1,2),
+                mybounds]
+        if guess is None:
+            guess = zeros(npts*2+2)
+        else:
+            guess = r_[guess[0].real,
+                    guess[1].real,
+                    guess[2:].view(float64)]
+        res = minimize(costfun, (guess,),
+                method='L-BFGS-B',
+                bounds=mybounds,
+                )
+        phcorr0,phcorr1,baseline_vec = vec_to_params(res.x)
+        baseline = generate_baseline(baseline_vec)
+        if show_plots: fl.plot(this_d*exp(-1j*phcorr1*d.fromaxis('t2')-1j*phcorr0)+baseline.C.ft('t2'),
+            label='after')
+        return phcorr0,phcorr1,baseline
+#loading data in
+if fl:
+    for exp_name,expno in [
+            ('w2_200309',2),
+            ]:
+        d = find_file(exp_name, exp_type='NMR_Data_AG', dimname = 'indirect', expno=expno)
+    print(ndshape(d))
+
+    print(d.getaxis('indirect'))
+    d.chunk('indirect',['indirect','ph1','ph2'],[-1,4,2]) #expands the indirect dimension into indirect, ph1, and ph2. inner most dimension is the inner most in the loop in pulse sequence, is the one on the farthest right. brackets with numbers are the number of phase cycle steps in each one. the number of steps is unknown in 'indirect' and is therefore -1.
+    print(d.getaxis('indirect'))
+    print(d.getaxis('ph1'))
+    print(d.getaxis('ph2'))
```

Comment:
replace all print statements with logger statement in vim:
```
:%s/^\( *\)print(\(.*\))$/\1logger.info(strm(\2))/
:%s/strm(\("[^"]\+"\))/\1/g
:%s/strm(\([^,]\+\))/\1/g
```

Reply:
PLEASE DO NOT RESOLVE MY COMMENTS... if you think it's done, just mark as done, if it's obsolete, explain why here.

for example, here, this is a general issue, and the vim commands above should be applied to the other top-level scripts

Reply:
done for all test scripts

Reply:
Just with a quick look, I see the following:
```
proc_CPMG.py:        print((iteration, x, f, int(accepted)))
proc_CPMG.py:    print(ndshape(s))
proc_CPMG.py:    print(T2)
proc_CPMG.py:    print("T2:",T2,"s")
proc_echo_mw.py:    print("THIS IS THE SHAPE")
proc_echo_mw.py:    print(ndshape(s))
proc_Hahn_echoph.py:    print(ndshape(s))
proc_Hahn_echoph.py:    print(ndshape(ph0))
proc_Hahn_echoph.py:        print('phasing dimension as one')
proc_Hahn_echoph.py:        print("there is only one dimension left -- standard 1D zeroth order phasing")
proc_IR.py:    print(s.dimlabels)
proc_IR.py:    print("best shift is",best_shift)
proc_IR.py:    print(ndshape(ph0))
proc_IR.py:        print('phasing dimension as one')
proc_IR.py:        print("there is only one dimension left -- standard 1D zeroth order phasing")
proc_IR.py:print("T1:",T1,"s")
proc_IR_noecho.py:print("Running")
proc_IR_noecho.py:    print(max_val)
proc_IR_noecho.py:    print(shape(mybounds))
proc_IR_noecho.py:    print(mybounds)
proc_IR_noecho.py:print("Estimated T1 is:", est_T1,"s")
proc_IR_noecho.py:print("SFO1 is",sfo1)
proc_IR_noecho.py:print("SAVING FILE")
proc_IR_noecho.py:print("FILE SAVED")
proc_nutation.py:    print(ndshape(s))
proc_square_refl.py:    print(("initial guess at center frequency at %0.5f MHz"%(center_frq/1e6)))
proc_square_refl.py:    print(center_frq)
proc_square_refl.py:        print(("seems to be more than one pulse -- on starting at "
proc_square_refl.py:    print("before filter",refl_blip_ranges)
proc_square_refl.py:    print("after filter",refl_blip_ranges)
proc_square_refl.py:    print(("found center frequency at %0.5f MHz"%(center_frq/1e6)))
proc_square_refl.py:    #print(nddata(refl_blip_ranges))
```

Reply:
yes I have fixed just forgot to push. should be okay now